mkosi/__init__.py

```diff
@@ -773,6 +773,20 @@ def install_unified_kernel(state: MkosiState, roothash: Optional[str]) -> None:
             if not (state.staging / state.config.output_split_uki).exists():
                 shutil.copy(boot_binary, state.staging / state.config.output_split_uki)
 
+                # ukify will have signed the kernel image as well. Let's make sure we put the signed kernel
+                # image in the output directory instead of the unsigned one by reading it from the UKI.
+
+                import pefile  # type: ignore
+                pe = pefile.PE(boot_binary, fast_load=True)
+                linux = {s.Name.decode().strip("\0"): s for s in pe.sections}[".linux"]
+                run(["dd",
```

Comment:
Why shell out to dd if you can get the data with `pe.get_data()` and let python write out the bytes?

Reply:
Yes, please, let's not shell out unless really useful.

Reply:
This was done in 9ffcdac128c66935aa5d5a98633fa7498bce92d1.